app/models/asset.rb

```diff
@@ -242,12 +242,11 @@ def self.extract_asset_text(asset_id)
         file_path = fa.path
       end
 
-      unless Yomu.class_eval('@@server_pid')
-        Yomu.server(:text, nil)
-        sleep(5)
+      if !ENV['NO_TIKA_SERVER'] && Yomu.class_variable_get(:@@server_pid).nil?
+        Yomu.server(:text)
       end
-      y = Yomu.new file_path
 
+      y = Yomu.new file_path
       text_data = y.text
```

Comment:
Add `Yomu.kill_server!` to `ensure` block.

Reply:
Why we should kill server if it was not started? And why we should kill it if we decided to start it?